components/net/bluetooth_thread.rs

```diff
@@ -450,16 +451,10 @@ impl BluetoothManager {
                 Some(id) => id.clone(),
                 None => return drop(sender.send(Err(String::from(DEVICE_MATCH_ERROR)))),
             };
-            let mut services = HashSet::new();
-            options.get_filters()
-                   .iter()
-                   .map(|filter| filter.get_services().iter().map(|s| services.insert(s.clone())).collect::<Vec<_>>())
-                   .collect::<Vec<_>>();
-            options.get_optional_services()
-                   .get_services()
-                   .iter()
-                   .map(|s| services.insert(s.clone()))
-                   .collect::<Vec<_>>();
+            let mut services = options.get_services_set();
+            if let Some(servives_set) = self.allowed_services.get(&device_id) {
```

Comment:
maybe `get_mut` and `services_set |= services;`
then we don't need the `insert` line